packages/react-router/docs/api/withRouter.md

```diff
@@ -31,14 +31,24 @@ const ShowTheLocationWithRouter = withRouter(ShowTheLocation)
 
 #### Important Note
 
-If you are using `withRouter` to prevent updates from being blocked by `shouldComponentUpdate`, it is important that `withRouter` wraps the component that implements `shouldComponentUpdate`. For example, when using Redux:
+`withRouter` only injects router props whenever it itself rerender. It does _not_ automatically rerender on route transitions unless its parent component re-renders. If you are using `withRouter` to prevent updates from being blocked by `shouldComponentUpdate`, it is important that `withRouter` wraps the component that implements `shouldComponentUpdate`. For example, when using Redux:
```

Comment:
The first sentence here could do with some editing. I would probably prefer something like:
>#### Important Notes
> `withRouter` does not subscribe to location changes like React Redux's `connect` does for state changes. Instead, re-renders after location changes propagate out from the `<Router>` component. This means that `withRouter` does _not_ re-render on route transitions unless its parent component re-renders.
>
>If you are using...

Also, I switched "rerender" to "re-render". I'm not sure if one is more correct than the other, but "re-render" seems to be the more common usage (and is even used later in the second sentence).